### ISSUE TITLE
more uncraftable items

### ISSUE THREAD

**Issue by wsor4035** [mt-mods/homedecor_modpack#41]
* [ ] homedecor:glowlight_quarter_14
* [ ] homedecor:wood_table_large_square
* [ ] homedecor:glass_table_large_square
* [ ] homedecor:wall_lamp_on
* [ ] homedecor:bed_extended
* [ ] homedecor:toilet_open
* [ ] homedecor:kitchen_cabinet_colorable_with_drawers_steel
* [ ] homedecor:kitchen_cabinet_colorable_with_drawers_granite
* [ ] homedecor:kitchen_cabinet_colorable_with_drawers
* [ ] homedecor:desk_lamp_14
* [ ] homedecor:ceiling_lamp_14
* [ ] homedecor:glowlight_half_14
* [ ] homedecor:standing_lamp_14
* [ ] homedecor:plasma_ball_on
* [ ] homedecor:hanging_lantern_14
* [ ] homedecor:ceiling_lantern_14
* [ ] homedecor:table_lamp_14
* [ ] homedecor:ground_lantern_14
* [ ] homedecor:rope_light_on_ceiling_on
* [ ] homedecor:glowlight_small_cube_14
* [ ] homedecor:plasma_lamp_14
* [ ] homedecor:lattice_lantern_large_14
* [ ] homedecor:rope_light_on_floor_on
* [ ] homedecor:kitchen_cabinet_colorable_with_drawers_marble
* [ ] homedecor:lattice_lantern_small_14
total items: 673
ignored items: 281
missing items: 25

https://xkcd.com/1205/

generated by: 
```lua
minetest.register_chatcommand("hd_test", {
	description = "does stuff, if you cant figure it out, dont use it",
	func = function()
		local valid = {
			homedecor = true,
			building_blocks = true,
			fake_fire = true,
			itemframes = true,
			lavalamp = true,
		}

		local output = {}
		local hd_count, ig_count, missing_count = 0, 0, 0
		for k, v in pairs(minetest.registered_items) do
			if valid[k:split(":")[1]] then
				hd_count = hd_count + 1
				local crafts = minetest.get_all_craft_recipes(k)
				if not crafts then
					if v.groups and v.groups.not_in_creative_inventory == 1 then
						ig_count = ig_count + 1
					else
						missing_count = missing_count + 1
						table.insert(output, "* [ ] " .. k)
					end
				end
			end
		end
		table.insert(output, "total items: " .. hd_count)
		table.insert(output, "ignored items: " .. ig_count)
		table.insert(output, "missing items: " .. missing_count)
		minetest.chat_send_all(dump(output))
		minetest.log("error", dump(output))

		--minetest.get_all_craft_recipes(query item)
	end
})
```
it may be worth improving this to recursively look through crafts results to validate all node names as well, but see the xkcd, something i slapped together in a few minutes

**Comment by BuckarooBanzay** [comment]
your snippet is pretty neat, any objections if i use that somewhere else? :smirk:

**Comment by S-S-X** [comment]
But keep in mind that xkcd does not apply to hobbies so go on and keep working on it 😄

**Comment by SwissalpS** [comment]
some of these don't need a recipe. e.g. toilet_open -> craft toilet then click on it to open.

**Comment by BuckarooBanzay** [comment]
> some of these don't need a recipe. e.g. toilet_open -> craft toilet then click on it to open.

Then it should fall into the `not_in_creative_inventory` category i think..

**Comment by dacmot** [comment]
Note that this check can be made with [qa_block](https://github.com/minetest-mods/qa_block/blob/master/checks/unobtainable_items.lua). Also it only reports the *_14 items as being uncraftable.

**Comment by wsor4035** [comment]
toliet is craftable, it uses groups, probably why it wasnt found

**Comment by wsor4035** [comment]
homedecor:wood_table_large_square
 homedecor:glass_table_large_square
 dont seem to exist?

**Comment by wsor4035** [comment]
![image](https://user-images.githubusercontent.com/24964441/188193967-a87448d2-bb13-4a5e-b835-64e376b88746.png)
wall lamp

**Comment by wsor4035** [comment]
bed extended fixed https://github.com/mt-mods/homedecor_modpack/commit/a598fa3067e81babeb09f39e42d14a6d720383f4

![image](https://user-images.githubusercontent.com/24964441/188194219-1c66ea40-1d71-4b8d-a2da-ab30d17aae8f.png)

**Comment by wsor4035** [comment]
kitchen cabinets are covered by https://github.com/mt-mods/homedecor_modpack/issues/4#issuecomment-1082550588

**Comment by wsor4035** [comment]
![image](https://user-images.githubusercontent.com/24964441/188194878-89b2a509-9812-407c-9aa7-6e5b4dffe130.png)
desk lamp

**Comment by wsor4035** [comment]
tldr it seems most of these are rather cursed crafting paths

**Comment by ulamthelucky** [comment]
Is there any known reason why the Wall Lamp recipe above that shows up in I3 and Unified Inventory will not show in sfinv / Crafting Guide?

![image](https://user-images.githubusercontent.com/299530/209877007-7e28bddf-dada-4ef2-b7e4-aa06033838ec.png)

**Comment by wsor4035** [comment]
@ulamthelucky no idea, out of scope for this issue. probably a bug with sfinv if it works with the other two.

**Comment by Lazerbeak12345** [comment]
Craft conflict - partially noted, but now I get the _other_ option between these two (IE OP is now backwards, but still  a well known problem.)

My idea is we could do the vertical doors -> drawers -> half thingy instead. (This idea is mostly why I'm bothering to comment).

![image](https://github.com/mt-mods/homedecor_modpack/assets/22641188/26da0bce-123f-4a69-9595-b47964abe045)
![image](https://github.com/mt-mods/homedecor_modpack/assets/22641188/facd2c24-ce4c-4b4e-b159-ff88067bcc9c)

**Comment by monk-afk** [comment]
We faced this issue after switching inventory from Unified Inventory to SFINV.
To troubleshoot, a fresh 5.9-dev with only minetest_game, Homedecor plus dependants, and tried first with the sfinv included with minetest_game, and then UI.
[Screenshots](https://github.com/monk-afk/SquareOne/issues/8#issuecomment-2113686874)

The recipes are not returned from minetest.get_all_craft_recipes(item_name), regardless of which inventory manager is installed.

I also tried hard-coding the item definitions from the registered_nodes table `print(dump(minetest.registered_items[item_name]))`, also copied the functions to for light on/off, the recipes do show in the sfinv.

That's all i got so far

**Comment by Niklp09** [comment]
Confirmed sfinv doesn't show any recipes for Small Glowlight Cube (homedecor:glowlight_small_cube_14), works w/ UI on the same setup.

**Comment by wsor4035** [comment]
if i was to guess/take a stab in the dark, see https://github.com/mt-mods/homedecor_modpack/blob/fa3f7f816b18f9aab2df47d05d12d24964817a5e/homedecor_lighting/init.lua#L1886, its probably something to do with the crafts being registered to the alias that sfinv cant handle https://github.com/mt-mods/homedecor_modpack/blob/fa3f7f816b18f9aab2df47d05d12d24964817a5e/homedecor_lighting/init.lua#L1658-L1673

**Comment by wsor4035** [comment]
also the issue isnt sfinv, mtg_craftguide is what handles crafts - at least for mtg

**Comment by wsor4035** [comment]
verified my guess by adding `_14` to the crafts, bug is in `mtg_craftguide`, please file @monk-afk on the mientest game repo for them to handle aliases.

**Comment by monk-afk** [comment]
If the aliases are registered before the crafts registrations, the craft recipes work fine. Is this still a mtg issue? 

![image](https://github.com/mt-mods/homedecor_modpack/assets/37543743/42b46345-0649-4bc2-878e-bc1f5897c837)
